homeassistant/components/sonos/helpers.py

```diff
@@ -115,9 +115,10 @@ def _find_target_identifier(instance: Any, fallback_soco: SoCo | None) -> str |
 def hostname_to_uid(hostname: str) -> str:
     """Convert a Sonos hostname to a uid."""
     if hostname.startswith("Sonos-"):
-        baseuid = hostname.split("-")[1].replace(".local.", "")
+        baseuid = hostname.removeprefix("Sonos-")
     elif hostname.startswith("sonos"):
-        baseuid = hostname[5:].replace(".local.", "")
+        baseuid = hostname.removeprefix("sonos")
     else:
         raise ValueError(f"{hostname} is not a sonos device.")
+    baseuid = baseuid.removesuffix(".local.")
```

Comment:
Slight behavior change if `.local.` is not at the end

Reply:
06c53516a1f61735ffa4ac0aa80bf06475b55fab